homeassistant/components/rest/__init__.py

```diff
@@ -150,6 +151,7 @@ def create_rest_data_from_config(hass, config):
     resource_template = config.get(CONF_RESOURCE_TEMPLATE)
     method = config.get(CONF_METHOD)
     payload = config.get(CONF_PAYLOAD)
```

Comment:
Why not detected if payload is a template instead? That way, we don't need another parameter.

Reply:
I've seen `xyz` and `xyz_template` quite often in HA, though -- like `resource_template` in this file. If that's indeed the convention, I'd say we should follow it (principle of least astonishment), even if it's an extra template.

Reply:
I think we should avoid that when possible. We have been removing it on several places in general as well. Most notability the `data` and `data_template` in ALL service calls (which now is just `data`).